library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/utils/Base64Utils.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.utils
+
+import android.graphics.Bitmap
+import android.os.Build
+import android.util.Base64
+import android.webkit.MimeTypeMap
+import androidx.annotation.VisibleForTesting
+import androidx.annotation.WorkerThread
+import java.io.ByteArrayOutputStream
+
+@VisibleForTesting
+internal const val BITMAP_SIZE_LIMIT_BYTES = 15000 // 15 kbs
+
+// This is the default compression for webp when writing to the output stream -
+// a lower quality leads to a lower filesize and worse fidelity image
+private const val IMAGE_QUALITY = 75
+
+@Suppress("UndocumentedPublicClass")
+object Base64Utils {
```

Comment:
Maybe we could clean up this file by having an interface
 
```
interface ImageCompression {
    fun compressBitmapToStream(bitmap: Bitmap): ByteArrayOutputStream
    fun isOverSizeLimit(bitmapSize: Int)
    fun getMimeType(): String?
}
```

And then have a default `WebPImageCompression` implementation

Reply:
Yes, this looks like a good approach. I've refactored to extract the methods